tests/tests_unit/test_api/test_assets.py

```diff
@@ -239,31 +239,40 @@ def test_update_multiple(self, mock_assets_response):
         assert isinstance(res, AssetList)
         assert mock_assets_response.calls[0].response.json()["items"] == res.dump(camel_case=True)
 
-    @pytest.mark.usefixtures("disable_gzip")
     def test_update_labels(self, mock_assets_response):
         ASSETS_API.update([AssetUpdate(id=1).labels.add(["PUMP", "ROTATING_EQUIPMENT"]).labels.remove(["VALVE"])])
-        request_body = json.loads(mock_assets_response.calls[0].request.body)["items"][0]["update"]
         expected = {
             "labels": {
                 "add": [{"externalId": "PUMP"}, {"externalId": "ROTATING_EQUIPMENT"}],
                 "remove": [{"externalId": "VALVE"}],
             }
         }
-        assert request_body == expected
+        assert expected == jsgz_load(mock_assets_response.calls[0].request.body)["items"][0]["update"]
 
-    @pytest.mark.usefixtures("disable_gzip")
-    def test_update_labels_using_deprecated_methods(self, mock_assets_response):
-        ASSETS_API.update([AssetUpdate(id=1).add_label("PUMP")])
-        request_body = json.loads(mock_assets_response.calls[0].request.body)["items"][0]["update"]
-        expected = {"labels": {"add": [{"externalId": "PUMP"}]}}
-        assert request_body == expected
-
-    @pytest.mark.usefixtures("disable_gzip")
+    # resource.update doesn't support full replacement of labels (set operation)
     def test_ignore_labels_resource_class(self, mock_assets_response):
-        ASSETS_API.update(Asset(id=1, labels=[{"external_id": "PUMP"}], name="Abc"))
-        request_body = json.loads(mock_assets_response.calls[0].request.body)["items"][0]["update"]
-        expected = {"name": {"set": "Abc"}}
-        assert request_body == expected
+        ASSETS_API.update(Asset(id=1, labels=[Label(external_id="Pump")], name="Abc"))
+        assert {"name": {"set": "Abc"}} == jsgz_load(mock_assets_response.calls[0].request.body)["items"][0]["update"]
+
+    def test_labels_filter_contains_all(self, mock_assets_response):
+        my_label_filter = LabelFilter(contains_all=["PUMP", "VERIFIED"])
```

Comment:
likewise here, does it support dict?
also what happens when you give both contains_all and contains_any?

Reply:
Doesn't support dict... I am not sure if the user will ever use the dict

Reply:
The api should return 400 in that case

Reply:
but filter does support dict, so isn't that inconsistent here? the main use is not having to import :)